engine/dlib/src/dlib/profile/profile.cpp

```diff
@@ -14,52 +14,563 @@
 
 #include "profile.h"
 #include <stdio.h>
+#include <stdio.h> // vsnprintf
+#include <stdarg.h> // va_start et al
 
-namespace dmProfile
+#include <dlib/atomic.h>
+#include <dlib/dstrings.h>
+#include <dlib/log.h>
+#include <dlib/mutex.h>
+
+struct ProfileProperty
+{
+    const char*             m_Name;
+    const char*             m_Desc;
+    ProfilePropertyValue    m_DefaultValue;
+    uint64_t                m_NameHash;
+    ProfileIdx              m_Idx;
+    ProfileIdx              m_ParentIdx;
+    ProfileIdx              m_FirstChild;
+    ProfileIdx              m_Sibling;
+    uint32_t                m_Flags;
+    ProfilePropertyType     m_Type;
+};
```

Comment:
profile.cpp is the base functionality, but not the profiler itself.
It passes on all the callbacks to its listeners.

Due to how Properties are setup, the base implementation needs to keep track of them, thus the ProfileProperty is a bit duplicated in the profiler_basic.cpp.